File: frontend/src/lib/api/system.ts
import { apiRequest } from "@/lib/api/client";

export interface BackendHealth {
  service: string;
  status: string;
  version: string;
  api_prefix: string;
  timestamp: string;
  allowed_origins: string[];
  playwright: {
    browser: string;
    executable_path: string | null;
    installed: boolean;
  };
}

export function getBackendHealth(): Promise<BackendHealth> {
  return apiRequest<BackendHealth>("/system/health");
}

File: backend/app/schemas/system.py
from datetime import datetime

from pydantic import BaseModel


class PlaywrightRuntime(BaseModel):
    browser: str
    executable_path: str | None
    installed: bool
    error: str | None = None


class HealthResponse(BaseModel):
    service: str
    status: str
    version: str
    api_prefix: str
    timestamp: datetime
    allowed_origins: list[str]
    playwright: PlaywrightRuntime
